conan/api/conan_api.py

```diff
@@ -60,6 +57,18 @@ def __init__(self, cache_folder=None):
         self.lockfile = LockfileAPI(self)
         self.local = LocalAPI(self)
 
-        required_range_new = self.config.global_conf.get("core:required_conan_version")
-        if required_range_new:
-            validate_conan_version(required_range_new)
+        _check_conan_version(self)
+
+    def reinit(self):
```

Comment:
Open to better names!

Reply:
Not specially opposed to ``reinit``, other possible alternatives
- reset
- restart
- invalidate_cache

Reply:
maybe `restart` looks good to me

Reply:
@AbrilRBS This was marked as solved but the same name remains. Will it be `reinit` in the end?